### PR TITLE
[SPARK-36357][SQL] Support pushdown Timestamp without time zone for orc

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR proposes to push down filters with `timestamp_ntz` to ORC.

### Why are the changes needed?
It's great to be able to push down filters with `timestamp_ntz` values for the better performance.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
New tests.

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
ping @gengliangwang @MaxGekk 
cc @cloud-fan

**Review by gengliangwang — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala [hunk 1]
```diff
@@ -168,6 +168,8 @@ private[sql] object OrcFilters extends OrcFiltersBase {
       toJavaDate(localDateToDays(value.asInstanceOf[LocalDate]))
     case _: TimestampType if value.isInstanceOf[Instant] =>
       toJavaTimestamp(instantToMicros(value.asInstanceOf[Instant]))
+    case _: TimestampNTZType =>
+      toJavaTimestamp(localDateTimeToMicros(value.asInstanceOf[LocalDateTime]))
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala:172]: shall we use `OrcUtils.toOrcNTZ`?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala [hunk 2]
```diff
@@ -168,6 +168,8 @@ private[sql] object OrcFilters extends OrcFiltersBase {
       toJavaDate(localDateToDays(value.asInstanceOf[LocalDate]))
     case _: TimestampType if value.isInstanceOf[Instant] =>
       toJavaTimestamp(instantToMicros(value.asInstanceOf[Instant]))
+    case _: TimestampNTZType =>
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala:171]:
```suggestion
    case _: TimestampNTZType if value.isInstanceOf[LocalDateTime] =>
```

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala (hunk 1 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala:172]: Hive not recognizes `OrcTimestamp`. Hive only have `Timestamp`.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@cloud-fan @gengliangwang Thank you for your review.

**Pull request closed without merging**